### DISCUSSION TITLE
Ability to add Images to Rich Text Editor?

### DISCUSSION THREAD

**Discussion by justinmckenzie** [apostrophecms/apostrophe#3231] [Q&A] [answered]
Hi, I was wondering if I could receive a code example, or be pointed in the direction of where I might be able to find one in the documentation (in v2), about how to add an "Image" option to `apostrophe-rich-text` `toolbar` array? 

My requirement is to be able to select an image icon from the rich text editor toolbar, which would allow for the user to select an image and add it inline to text/area that is currently being edited. Please let me know if there is any more helpful detail I can provide. Thank you in advance!

**Comment by boutell** [comment, marked as answer]
Hi Justin,

Apostrophe handles this differently, through image widgets, which are on
the "add content" menu (the "+" sign), so you can interleave them with rich
text widgets.

However there is a "split" control you can add to your rich text toolbar,
by that name, if you get into a situation where you need an image in the
middle of an existing rich text widget. This splits the rich text widget in
half at the cursor, allowing you to add a widget in between.

On Tue, Jul 13, 2021 at 3:06 PM Justin McKenzie ***@***.***>
wrote:

> Hi, I was wondering if I could receive a code example, or be pointed in
> the direction of where I might be able to find one in the documentation (in
> v2), about how to add an "Image" option to apostrophe-rich-text toolbar
> array?
>
> My requirement is to be able to select an image icon from the rich text
> editor toolbar, which would allow for the user to select an image and add
> it inline to text/area that is currently being edited. Please let me know
> if there is any more helpful detail I can provide. Thank you in advance!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/3231>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27KHLN3IVDAZQKG6VT3TXSFCNANCNFSM5AJ3DWLQ>
> .
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Reply by justinmckenzie** [reply]
Hey Tom, thanks for the clarification! So it sounds like I am not able to add an image directly into the "rich text widget" (in between text/paragraphs, etc), it would have to be split or added into a different widget between the text?

**Reply by abea** [reply]
That's correct.

**Comment by rscarpim1973** [comment]
Do we have any changes? Now we're going to be able to add an image inside a rich-text?

**Comment by boutell** [comment]
Image widgets are still added separately from rich text. There is a good chance we'll add a way to add one in the middle of a rich text widget, which would automatically split it at that point, but we don't have plans to change Apostrophe to directly incorporate images inside the rich text HTML.
